client/apps/edit/components/content/article_layouts/video.jsx

```diff
@@ -0,0 +1,234 @@
+import FileInput from '/client/components/file_input/index.jsx'
+import Paragraph from '/client/components/rich_text/components/paragraph.coffee'
+import { PlainText } from '/client/components/rich_text/components/plain_text'
+import { ProgressBar } from '/client/components/file_input/progress_bar.jsx'
+import { IconRemove } from '@artsy/reaction-force/dist/Components/Publishing/Icon/IconRemove'
+
+export class EditVideo extends Component {
+  static propTypes = {
+    article: PropTypes.object.isRequired,
+    onChange: PropTypes.func.isRequired
+  }
+
+  state = {
+    uploadProgress: null,
+    showVideo: false
+  }
+
+  onMediaChange = (key, value) => {
+    const { article, onChange } = this.props
+    const media = article.get('media') || {}
+
+    if (key !== 'duration' && key !== 'published') {
+      media[key] = value.length ? value : null
+    } else {
+      media[key] = value
+    }
+    onChange('media', media)
+  }
+
+  setDuration = (e) => {
+    this.onMediaChange('duration', e.target.duration)
+  }
+
+  toggleVideoPreview = (showVideo) => {
+    this.setState({ showVideo })
+  }
+
+  editDescription = () => {
+    const { article, onChange } = this.props
+
+    return (
+      <PlainText
+        content={article.get('description')}
+        onChange={(key, value) => onChange('description', value)}
+        placeholder='Description'
+        name='description'
+      />
+    )
+  }
+
+  editTitle = () => {
+    const { article, onChange } = this.props
+
+    return (
+      <PlainText
+        content={article.get('title')}
+        onChange={(key, value) => onChange('title', value)}
+        placeholder='Title'
+        name='title'
+      />
+    )
+  }
+
+  editMediaDescription = () => {
+    const { article } = this.props
+    const description = article.get('media') && article.get('media').description
+
+    return (
+      <Paragraph
+        html={description || ''}
+        linked
+        onChange={(html) => this.onMediaChange('description', html)}
+        placeholder='Start writing here...'
+      />
+    )
+  }
+
+  editMediaCredits = () => {
+    const { article } = this.props
+    const credits = article.get('media') && article.get('media').credits
+
+    return (
+      <Paragraph
+        html={credits || ''}
+        linked
+        onChange={(html) => this.onMediaChange('credits', html)}
+        placeholder='Start writing here...'
+      />
+    )
+  }
+
+  render () {
+    const { article } = this.props
+    const { uploadProgress } = this.state
+    const media = article.get('media') || {}
+
+    return (
+      <EditVideoContainer>
+        {uploadProgress &&
+          <ProgressBar progress={uploadProgress} />
+        }
+
+        <VideoPreview visible={this.state.showVideo}>
+          <VideoPlayer
+            url={media.url}
+            onLoadedMetadata={this.setDuration}
+          />
```

Comment:
setDuration and toggleVideoPreview are declared as component methods but could easily be set inline like on 131 and 133, for consistencies sake.